packages/vscode-extension/src/debugging/MyBreakpoint.ts

```diff
@@ -0,0 +1,135 @@
+import { Breakpoint, Source } from "@vscode/debugadapter";
+import { CDPCommunicatorInterface } from "./CDPCommunicator";
+import { SourceMapController } from "./SourceMapsController";
+import { Logger } from "../Logger";
+
+export class MyBreakpoint extends Breakpoint {
+  public readonly line: number;
+  public readonly column: number | undefined;
+  private _id: number | undefined;
+  private executingQueue: boolean = false;
+  private cdpCommunicationQueue: Array<() => Promise<void>> = [];
+
+  constructor(
+    private CDPCommunicator: CDPCommunicatorInterface,
+    private sourceMapController: SourceMapController,
+    verified: boolean,
+    line: number,
+    column?: number,
+    source?: Source
+  ) {
+    super(verified, line, column, source);
+    this.column = column;
+    this.line = line;
+  }
+
+  public async delete() {
+    return this.addTaskToQueue(async () => {
+      await this.deleteCDPBreakpoint();
+    });
+  }
+
+  public reset(sourceMapPath: string) {
+    return this.addTaskToQueue(async () => {
+      await this.resetCDPBreakpoint(sourceMapPath);
+    });
```

Comment:
shouldn't this just call `delete` and then `add` ? it seems like that's exactly what is happening under the hood.

Reply:
almost but we additionally change the verification status of the breakpoint to false after deleting it

```
if (this.verified) {
      await this.cdpSession.sendCDPMessage("Debugger.removeBreakpoint", {
        breakpointId: this.getId(),
      });
      this.verified = false; // <- here is the difference 
    }
    await this.setCDPBreakpoint(sourceMapPath);
```

It is usually unnecessary to do that after deleting

I'll change the code to: 

```
private async resetCDPBreakpoint(sourceMapPath: string) {
    if (this.verified) {
      await this.deleteCDPBreakpoint();
      this.verified = false;
    }
    await this.setCDPBreakpoint(sourceMapPath);
  }
``` 
to make the difference clearer